src/kubernetes_cluster/spec/pod_event/state_machine.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use super::types::*;
+use crate::external_api::spec::*;
+use crate::kubernetes_api_objects::spec::{pod::*, resource::*};
+use crate::kubernetes_cluster::spec::{cluster::Cluster, message::*};
+use crate::reconciler::spec::reconciler::Reconciler;
+use crate::state_machine::action::*;
+use crate::state_machine::state_machine::*;
+use crate::temporal_logic::defs::*;
+use vstd::{multiset::*, prelude::*};
+
+verus! {
+
+impl <K: CustomResourceView, E: ExternalAPI, R: Reconciler<K, E>> Cluster<K, E, R> {
+
+pub open spec fn create_pod() -> PodEventAction<E::Input, E::Output> {
+    Action {
+        precondition: |input: PodEventActionInput, s: PodEventState| {
```

Comment:
@codyjrivera we can remove the precondition of this and other actions (just make it `true`). We don't care whether the pod_event issues a valid pod or not as we don't need it to make progress.